### PR TITLE
Start proj

### PR DESCRIPTION
**Pull request by KrlSed**

### CODE REVIEW

**Comment by KrlSed** [comment]
1

File: pom.xml [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+            <artifactId>testcontainers</artifactId>
+            <version>1.17.3</version>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.mapstruct</groupId>
+            <artifactId>mapstruct</artifactId>
+            <version>1.4.2.Final</version>
+        </dependency>
+    </dependencies>
```
Comment (borecjudo2) [inline pom.xml]: My suggestion to use dependencyManager for version of dependencies

File: src/main/java/com/sedliarov/learningtable/controller/GroupController.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@RestController
+@RequestMapping("/groups")
+public class GroupController {
+    @Autowired
+    private GroupRepository repository;
```
Comment (borecjudo2) [inline src/main/java/com/sedliarov/learningtable/controller/GroupController.java]: Learn about @AllArgsConstructor and constructor's bean injection in Spring. My proposal to you

File: src/main/java/com/sedliarov/learningtable/controller/GroupController.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+    @PostMapping
+    public Group addGroup(@RequestBody Group group)
+    {
+        return repository.save(group);
+    }
+
+    @GetMapping()
+    public List<Group> getGroups() {return repository.findAll();}
+
```
Comment (borecjudo2) [inline src/main/java/com/sedliarov/learningtable/controller/GroupController.java]: I'm stressed with seeing this code style. You can use IDE code style, use keys combination ( cntr+alt+O cntr+alt+L)

Comment (borecjudo2) [reply src/main/java/com/sedliarov/learningtable/controller/GroupController.java]: For all files

File: src/main/java/com/sedliarov/learningtable/controller/GroupController.java (hunk 2 above)
Comment (borecjudo2) [inline src/main/java/com/sedliarov/learningtable/controller/GroupController.java]: Controller can work with service and service with repo and no other way

File: src/main/java/com/sedliarov/learningtable/controller/GroupController.java [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    @Autowired
+    private GroupRepository repository;
+
+    @PostMapping
+    public Group addGroup(@RequestBody Group group)
+    {
+        return repository.save(group);
+    }
+
+    @GetMapping()
```
Comment (borecjudo2) [inline src/main/java/com/sedliarov/learningtable/controller/GroupController.java]:
```suggestion
    @GetMapping
```

Comment (borecjudo2) [reply src/main/java/com/sedliarov/learningtable/controller/GroupController.java]: Also look to   @ResponseStatus(HttpStatus.OK) and MediaType of endpoint

File: src/main/java/com/sedliarov/learningtable/controller/GroupController.java [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+                .map(group -> {
+                    group.setName(newGroup.getName());
+                    group.setTeacher(newGroup.getTeacher());
+                    group.setStudentList(newGroup.getStudentList());
+                    return repository.save(group);
+                })
+                .orElseGet(() -> {
+                    newGroup.setId(id);
+                    return repository.save(newGroup);
+                });
```
Comment (borecjudo2) [inline src/main/java/com/sedliarov/learningtable/controller/GroupController.java]: To service layer

File: src/main/java/com/sedliarov/learningtable/dto/GroupDto.java [hunk 6]
```diff
@@ -0,0 +1,65 @@
+package com.sedliarov.learningtable.dto;
```
Comment (borecjudo2) [inline src/main/java/com/sedliarov/learningtable/dto/GroupDto.java]:
```suggestion
package com.sedliarov.learningtable.model.dto;
```

Comment (borecjudo2) [reply src/main/java/com/sedliarov/learningtable/dto/GroupDto.java]: and the same for other

File: src/main/java/com/sedliarov/learningtable/model/Student.java [hunk 7]
```diff
@@ -0,0 +1,30 @@
+package com.sedliarov.learningtable.model;
```
Comment (borecjudo2) [inline src/main/java/com/sedliarov/learningtable/model/Student.java]:
```suggestion
package com.sedliarov.learningtable.model.entity;
```

Comment (borecjudo2) [reply src/main/java/com/sedliarov/learningtable/model/Student.java]: the same for other

File: src/main/java/com/sedliarov/learningtable/service/GroupService.java [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+import com.sedliarov.learningtable.model.Group;
+
+import java.util.List;
+import java.util.UUID;
+
+public interface GroupService {
+
+    GroupDto addGroup(GroupDto groupDto);
+
+    GroupDto saveGroup(UUID id, GroupDto groupDto);
```
Comment (borecjudo2) [inline src/main/java/com/sedliarov/learningtable/service/GroupService.java]: Why we have id for save method?

File: src/main/resources/application.properties [hunk 9]
```diff
@@ -0,0 +1,4 @@
+spring.datasource.driver-class-name=org.postgresql.Driver
```
Comment (borecjudo2) [inline src/main/resources/application.properties]: Take a look about yml

File: src/main/java/com/sedliarov/learningtable/dto/GroupDto.java [hunk 10]
```diff
@@ -0,0 +1,65 @@
...
+
+import javax.persistence.CascadeType;
+import javax.persistence.JoinColumn;
+import javax.persistence.OneToMany;
+import javax.persistence.OneToOne;
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+
```
Comment (borecjudo2) [inline src/main/java/com/sedliarov/learningtable/dto/GroupDto.java]: Use needed Lombok annotation for all DTOs

**Review by borecjudo2 — COMMENTED** [review]
For first review it will be enough. Solve the issues and come for the new one review)

File: pom.xml (hunk 1 above)
Comment (KrlSed) [reply pom.xml]: Done

File: src/main/java/com/sedliarov/learningtable/controller/GroupController.java (hunk 2 above)
Comment (KrlSed) [reply src/main/java/com/sedliarov/learningtable/controller/GroupController.java]: Done

File: src/main/java/com/sedliarov/learningtable/controller/GroupController.java (hunk 3 above)
Comment (KrlSed) [reply src/main/java/com/sedliarov/learningtable/controller/GroupController.java]: Done

File: src/main/java/com/sedliarov/learningtable/controller/GroupController.java (hunk 2 above)
Comment (KrlSed) [reply src/main/java/com/sedliarov/learningtable/controller/GroupController.java]: Done

File: src/main/java/com/sedliarov/learningtable/controller/GroupController.java (hunk 4 above)
Comment (KrlSed) [reply src/main/java/com/sedliarov/learningtable/controller/GroupController.java]: Done

File: src/main/java/com/sedliarov/learningtable/dto/GroupDto.java (hunk 6 above)
Comment (KrlSed) [reply src/main/java/com/sedliarov/learningtable/dto/GroupDto.java]: Done

File: src/main/java/com/sedliarov/learningtable/model/Student.java (hunk 7 above)
Comment (KrlSed) [reply src/main/java/com/sedliarov/learningtable/model/Student.java]: Done

File: src/main/java/com/sedliarov/learningtable/service/GroupService.java (hunk 8 above)
Comment (KrlSed) [reply src/main/java/com/sedliarov/learningtable/service/GroupService.java]: Done

File: src/main/resources/application.properties (hunk 9 above)
Comment (KrlSed) [reply src/main/resources/application.properties]: Done

File: src/main/java/com/sedliarov/learningtable/dto/GroupDto.java (hunk 10 above)
Comment (KrlSed) [reply src/main/java/com/sedliarov/learningtable/dto/GroupDto.java]: Done

File: pom.xml [hunk 11]
```diff
@@ -0,0 +1,151 @@
...
+    <description>learning-table</description>
+
+    <properties>
+        <java.version>17</java.version>
+        <mapstruct.version>1.4.2.Final</mapstruct.version>
+        <testcontainers.version>1.17.3</testcontainers.version>
+        <postgrees.version>42.5.0</postgrees.version>
+        <hibernate-core.version>6.1.0.Final</hibernate-core.version>
+
+    </properties>
```
Comment (borecjudo2) [inline pom.xml]:
```suggestion
        <hibernate-core.version>6.1.0.Final</hibernate-core.version>
    </properties>
```

File: pom.xml [hunk 12]
```diff
@@ -0,0 +1,151 @@
...
+                </configuration>
+            </plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-compiler-plugin</artifactId>
+                <version>3.8.1</version>
+                <configuration>
+                    <source>1.8</source> <!-- depending on your project -->
+                    <target>1.8</target> <!-- depending on your project -->
+                    <annotationProcessorPaths>
```
Comment (borecjudo2) [inline pom.xml]:
```suggestion
                <configuration>
                    <source>17</source> <!-- depending on your project -->
                    <target>17</target> <!-- depending on your project -->
                    <annotationProcessorPaths>
```

File: pom.xml [hunk 13]
```diff
@@ -0,0 +1,151 @@
...
+                        </path>
+                        <path>
+                            <groupId>org.projectlombok</groupId>
+                            <artifactId>lombok</artifactId>
+                            <version>${lombok.version}</version>
+                        </path>
+                        <path>
+                            <groupId>org.projectlombok</groupId>
+                            <artifactId>lombok-mapstruct-binding</artifactId>
+                            <version>0.2.0</version>
```
Comment (borecjudo2) [inline pom.xml]: version to properties

Comment (borecjudo2) [inline pom.xml]: version to porperties

File: src/main/java/com/sedliarov/learningtable/controller/GroupController.java [hunk 14]
```diff
@@ -0,0 +1,48 @@
...
+    }
+
+    @DeleteMapping("/{id}")
+    @ResponseStatus(HttpStatus.NO_CONTENT)
+    public String deleteGroup(@PathVariable int id) {
+        service.deleteGroup(id);
+        return "Group " + id + " has been deleted";
+    }
+
+}
```
Comment (borecjudo2) [inline src/main/java/com/sedliarov/learningtable/controller/GroupController.java]:
```suggestion
    @PostMapping
    @ResponseStatus(HttpStatus.CREATED)
    public GroupDto addGroup(@RequestBody GroupDto group) {
        return service.addGroup(group);
    }

    @PutMapping("/{id}")
    @ResponseStatus(HttpStatus.NO_CONNTENT)
    public GroupDto updateGroup(@RequestBody GroupDto group, @PathVariable Integer groupId) {
        return service.updateGroup(group, groupId);
    }
    
    @DeleteMapping("/{id}")
    @ResponseStatus(HttpStatus.OK)
    public String deleteGroup(@PathVariable Integer groupId) {
        service.deleteGroup(groupId);
    }

    @GetMapping("/{id}")
    @ResponseStatus(HttpStatus.OK)
    public GroupDto getGroup(@PathVariable Integer groupId) {
        return service.getGroupById(groupId);
    }
    
    @GetMapping
    @ResponseStatus(HttpStatus.OK)
    public List<GroupDto> getGroups() {
        return service.getGroups();
    }
}

```

Comment (borecjudo2) [reply src/main/java/com/sedliarov/learningtable/controller/GroupController.java]: The same for other controllers

File: src/main/java/com/sedliarov/learningtable/model/dto/GroupDto.java [hunk 15]
```diff
@@ -0,0 +1,21 @@
...
+import lombok.Data;
+
+import java.io.Serializable;
+import java.util.List;
+
+@Data
+public class GroupDto implements Serializable {
+
+    private int id;
+
```
Comment (borecjudo2) [inline src/main/java/com/sedliarov/learningtable/model/dto/GroupDto.java]:
```suggestion
    private UUID id;

```

Comment (borecjudo2) [reply src/main/java/com/sedliarov/learningtable/model/dto/GroupDto.java]: For the entities and all DTOs

File: src/main/java/com/sedliarov/learningtable/model/entity/Student.java [hunk 16]
```diff
@@ -0,0 +1,31 @@
...
+    @Id
+    @GeneratedValue
+    private int id;
+
+    private String firstName;
+
+    private String secondName;
+
+    private Double note;
+}
```
Comment (borecjudo2) [inline src/main/java/com/sedliarov/learningtable/model/entity/Student.java]:
```suggestion
    private String secondName;

    private Double note;
    
    @ManyToOne
    @JoinColumn(name="group_id", nullable=false)
    private Group group;
}
```

Comment (borecjudo2) [inline src/main/java/com/sedliarov/learningtable/model/entity/Student.java]:
```suggestion
public class Student{

    @Id
    @GeneratedValue
    private UUID studentId;

    private String firstName;

    private String secondName;

    private Double note;
}
```

File: src/main/java/com/sedliarov/learningtable/model/entity/Group.java [hunk 17]
```diff
@@ -0,0 +1,33 @@
...
+
+    private String name;
+
+    @OneToOne
+    @JoinColumn
+    private Teacher teacher;
+
+    @OneToMany(cascade = CascadeType.ALL)
+    @JoinColumn
+    private List<Student> studentList = new ArrayList<>();
```
Comment (borecjudo2) [inline src/main/java/com/sedliarov/learningtable/model/entity/Group.java]:
```suggestion
public class Group {

    @Id
    @GeneratedValue
    private UUID groupId;

    private String name;

    @OneToOne
    @JoinColumn(name = "teacher_id", referencedColumnName = "id")
    private Teacher teacher;

    @OneToMany(mappedBy="group")
    private Set<Student> students;
```

File: src/main/java/com/sedliarov/learningtable/repository/GroupRepository.java [hunk 18]
```diff
@@ -0,0 +1,8 @@
+package com.sedliarov.learningtable.repository;
+
+import com.sedliarov.learningtable.model.entity.Group;
+import org.springframework.data.jpa.repository.JpaRepository;
+
+
+public interface GroupRepository extends JpaRepository<Group, Integer>{
+}
```
Comment (borecjudo2) [inline src/main/java/com/sedliarov/learningtable/repository/GroupRepository.java]: And the same for other
```suggestion
public interface GroupRepository extends JpaRepository<Group, UUID>{
}
```

File: src/main/java/com/sedliarov/learningtable/service/GroupService.java [hunk 19]
```diff
@@ -0,0 +1,18 @@
...
+
+import com.sedliarov.learningtable.model.dto.GroupDto;
+
+import java.util.List;
+
+public interface GroupService {
+
+    GroupDto addGroup(GroupDto groupDto);
+
+    GroupDto updateGroup(GroupDto groupDto);
```
Comment (borecjudo2) [inline src/main/java/com/sedliarov/learningtable/service/GroupService.java]: Take a look on controller comments

File: src/main/java/com/sedliarov/learningtable/service/impl/GroupServiceImpl.java [hunk 20]
```diff
@@ -0,0 +1,63 @@
...
+    public GroupDto addGroup(GroupDto groupDto) {
+        try {
+            Group dtoToGroup = mapper.dtoToGroup(groupDto);
+            Group group = repository.save(dtoToGroup);
+            return mapper.groupToDto(group);
+        }catch (Exception e) {
+            e.printStackTrace();
+            return null;
+        }
+    }
```
Comment (borecjudo2) [inline src/main/java/com/sedliarov/learningtable/service/impl/GroupServiceImpl.java]:
```suggestion
    public GroupDto addGroup(GroupDto groupDto) {
        Group groupToSave= mapper.dtoToGroup(groupDto);
         return mapper.groupToDto(repository.save(groupToSave));
    }
```

File: src/main/java/com/sedliarov/learningtable/service/impl/GroupServiceImpl.java [hunk 21]
```diff
@@ -0,0 +1,63 @@
...
+        return mapper.groupToDto(repository.findById(groupDto.getId()).map(group -> {
+            group.setName(groupDto.getName());
+            group.setTeacher(groupDto.getTeacher());
+            group.setStudentList(groupDto.getStudentList());
+            return repository.save(group);
+        }).orElseGet(() -> {
+            groupDto.setId(groupDto.getId());
+            return repository.save(mapper.dtoToGroup(groupDto));
+        }));
+    }
```
Comment (borecjudo2) [inline src/main/java/com/sedliarov/learningtable/service/impl/GroupServiceImpl.java]: What's going on here?

File: src/main/java/com/sedliarov/learningtable/service/impl/GroupServiceImpl.java [hunk 22]
```diff
@@ -0,0 +1,63 @@
...
+
+    @Override
+    public GroupDto getGroupById(int id) {
+        return mapper.groupToDto(repository.findById(id).orElseThrow(RuntimeException::new));
+    }
+
+    @Override
+    public List<GroupDto> getAllGroup() {
+        List<Group> groups = (List<Group>) repository.findAll();
+        return groups.stream().map(mapper::groupToDto).collect(Collectors.toList());
```
Comment (borecjudo2) [inline src/main/java/com/sedliarov/learningtable/service/impl/GroupServiceImpl.java]:
```suggestion
        return repository.findAll().stream().map(mapper::groupToDto).toList();
```

File: src/main/java/com/sedliarov/learningtable/service/impl/GroupServiceImpl.java [hunk 23]
```diff
@@ -0,0 +1,63 @@
...
+
+    @Override
+    public void deleteGroup(int id) {
+        repository.deleteById(id);
+    }
+
+    @Override
+    public GroupDto getGroupById(int id) {
+        return mapper.groupToDto(repository.findById(id).orElseThrow(RuntimeException::new));
+    }
```
Comment (borecjudo2) [inline src/main/java/com/sedliarov/learningtable/service/impl/GroupServiceImpl.java]:
```suggestion
    @Override
    public GroupDto getGroupById(int id) {
        return mapper.groupToDto(repository.findById(id).orElseThrow(EntityNotFoundException::new));
    }
```

File: src/main/resources/application.yml [hunk 24]
```diff
@@ -0,0 +1,11 @@
...
+  jpa:
+    hibernate:
+      ddl-auto: create
+  datasource:
+    driver-class-name: org.testcontainers.jdbc.ContainerDatabaseDriver
+    url: jdbc:tc:postgresql:9.6.8:///test
+    username: test
+    password: test
+
+
```
Comment (borecjudo2) [inline src/main/resources/application.yml]:
```suggestion
spring:
  jpa:
    hibernate:
      ddl-auto: create
  datasource:
    driver-class-name: org.testcontainers.jdbc.ContainerDatabaseDriver
    url: jdbc:tc:postgresql:9.6.8:///test
    username: test
    password: test

```

File: src/main/java/com/sedliarov/learningtable/mapper/GroupMapper.java [hunk 25]
```diff
@@ -0,0 +1,15 @@
...
+import org.mapstruct.Mapper;
+import org.mapstruct.factory.Mappers;
+
+@Mapper
+public interface GroupMapper {
+    GroupMapper INSTANCE = Mappers.getMapper(GroupMapper.class);
+
+    Group dtoToGroup(GroupDto groupDto);
+
+    GroupDto groupToDto(Group group);
```
Comment (borecjudo2) [inline src/main/java/com/sedliarov/learningtable/mapper/GroupMapper.java]:
```suggestion
    Group dtoToEntity(GroupDto groupDto);

    GroupDto entityToDto(Group group);
```

Comment (borecjudo2) [reply src/main/java/com/sedliarov/learningtable/mapper/GroupMapper.java]: The same for other

Comment (KrlSed) [reply src/main/java/com/sedliarov/learningtable/mapper/GroupMapper.java]: Done

File: src/main/java/com/sedliarov/learningtable/service/impl/GroupServiceImpl.java (hunk 23 above)
Comment (KrlSed) [reply src/main/java/com/sedliarov/learningtable/service/impl/GroupServiceImpl.java]: done

File: src/main/java/com/sedliarov/learningtable/service/impl/GroupServiceImpl.java (hunk 22 above)
Comment (KrlSed) [reply src/main/java/com/sedliarov/learningtable/service/impl/GroupServiceImpl.java]: Done

File: src/main/java/com/sedliarov/learningtable/service/impl/GroupServiceImpl.java (hunk 21 above)
Comment (KrlSed) [reply src/main/java/com/sedliarov/learningtable/service/impl/GroupServiceImpl.java]: Done

File: src/main/java/com/sedliarov/learningtable/controller/GroupController.java (hunk 14 above)
Comment (KrlSed) [reply src/main/java/com/sedliarov/learningtable/controller/GroupController.java]: done

File: pom.xml (hunk 13 above)
Comment (KrlSed) [reply pom.xml]: done

Comment (KrlSed) [reply pom.xml]: done

File: pom.xml (hunk 12 above)
Comment (KrlSed) [reply pom.xml]: done

File: pom.xml (hunk 11 above)
Comment (KrlSed) [reply pom.xml]: done

File: pom.xml [hunk 26]
```diff
@@ -0,0 +1,150 @@
...
+                <groupId>org.postgresql</groupId>
+                <artifactId>postgresql</artifactId>
+                <version>${postgrees.version}</version>
+            </dependency>
+            <dependency>
+                <groupId>org.hibernate</groupId>
+                <artifactId>hibernate-core</artifactId>
+                <version>${hibernate-core.version}</version>
+                <type>pom</type>
+            </dependency>
```
Comment (borecjudo2) [inline pom.xml]: Why we need this one?

File: pom.xml [hunk 27]
```diff
@@ -0,0 +1,150 @@
...
+            <artifactId>spring-boot-test</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.postgresql</groupId>
+            <artifactId>postgresql</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.hibernate</groupId>
+            <artifactId>hibernate-core</artifactId>
+        </dependency>
```
Comment (borecjudo2) [inline pom.xml]: And this

File: src/main/java/com/sedliarov/learningtable/controller/GroupController.java [hunk 28]
```diff
@@ -0,0 +1,49 @@
...
+    public GroupDto getGroup(@PathVariable UUID id) {
+        return service.getGroupById(id);
+    }
+
+    @DeleteMapping("/{id}")
+    @ResponseStatus(HttpStatus.OK)
+    public String deleteGroup(@PathVariable UUID id) {
+        service.deleteGroup(id);
+        return "Group " + id + " has been deleted";
+    }
```
Comment (borecjudo2) [inline src/main/java/com/sedliarov/learningtable/controller/GroupController.java]:
```suggestion
    @DeleteMapping("/{id}")
    @ResponseStatus(HttpStatus.OK)
    public void deleteGroup(@PathVariable UUID id) {
        service.deleteGroup(id);
    }
```

Comment (borecjudo2) [reply src/main/java/com/sedliarov/learningtable/controller/GroupController.java]: And for other

File: src/main/java/com/sedliarov/learningtable/controller/HelloController.java [hunk 29]
```diff
@@ -0,0 +1,17 @@
...
+    @GetMapping("/hello")
+    public String sayHello() {
+        return "Hellooooooo";
+    }
+
+    @GetMapping("/create")
+    public String sayHi() {
+        return "Create";
+    }
+}
```
Comment (borecjudo2) [inline src/main/java/com/sedliarov/learningtable/controller/HelloController.java]: Need to delete

File: src/main/java/com/sedliarov/learningtable/model/entity/Group.java [hunk 30]
```diff
@@ -0,0 +1,32 @@
...
+    private UUID groupId;
+
+    private String name;
+
+    @OneToOne
+    @JoinColumn(name = "teacherId", referencedColumnName = "id")
+    private Teacher teacher;
+
+    @OneToMany(mappedBy="studentId")
+    private Set<Student> students;
```
Comment (borecjudo2) [inline src/main/java/com/sedliarov/learningtable/model/entity/Group.java]: Are you sure with correctly value in mappedBy?

File: src/main/java/com/sedliarov/learningtable/model/entity/Student.java [hunk 31]
```diff
@@ -0,0 +1,33 @@
...
+import javax.persistence.*;
+import java.util.UUID;
+
+@Entity
+@Table(name = "students")
+@Getter
+@Setter
+@AllArgsConstructor
+@NoArgsConstructor
+public class Student{
```
Comment (borecjudo2) [inline src/main/java/com/sedliarov/learningtable/model/entity/Student.java]:
```suggestion
public class Student {
```

File: src/main/java/com/sedliarov/learningtable/repository/GroupRepository.java [hunk 32]
```diff
@@ -0,0 +1,12 @@
...
+
+import com.sedliarov.learningtable.model.entity.Group;
+import org.springframework.data.jpa.repository.JpaRepository;
+
+import java.util.Optional;
+import java.util.UUID;
+
+
+public interface GroupRepository extends JpaRepository<Group, UUID>{
+    Optional<Group> findById(UUID id);
```
Comment (borecjudo2) [inline src/main/java/com/sedliarov/learningtable/repository/GroupRepository.java]: We really need this one?

File: src/main/java/com/sedliarov/learningtable/model/entity/Teacher.java [hunk 33]
```diff
@@ -0,0 +1,33 @@
...
+import javax.persistence.Table;
+import java.util.UUID;
+
+@Entity
+@Table(name = "teachers")
+@Getter
+@Setter
+@AllArgsConstructor
+@NoArgsConstructor
+public class Teacher{
```
Comment (borecjudo2) [inline src/main/java/com/sedliarov/learningtable/model/entity/Teacher.java]:
```suggestion
public class Teacher {
```

File: src/main/java/com/sedliarov/learningtable/service/GroupService.java [hunk 34]
```diff
@@ -0,0 +1,19 @@
+package com.sedliarov.learningtable.service;
+
+import com.sedliarov.learningtable.model.dto.GroupDto;
+
+import java.util.List;
+import java.util.UUID;
+
+public interface GroupService {
+
+    GroupDto addGroup(GroupDto groupDto);
```
Comment (borecjudo2) [inline src/main/java/com/sedliarov/learningtable/service/GroupService.java]:
```suggestion
    GroupDto createGroup(GroupDto groupDto);
```

Comment (borecjudo2) [reply src/main/java/com/sedliarov/learningtable/service/GroupService.java]: And the same for controller layer and for other entities

File: src/main/java/com/sedliarov/learningtable/service/GroupService.java [hunk 35]
```diff
@@ -0,0 +1,19 @@
...
+
+    GroupDto updateGroup(UUID id, GroupDto groupDto);
+
+    void deleteGroup(UUID id);
+
+    GroupDto getGroupById(UUID id);
+
+    List<GroupDto> getAllGroup();
```
Comment (borecjudo2) [inline src/main/java/com/sedliarov/learningtable/service/GroupService.java]:
```suggestion
    List<GroupDto> getGroups();
```

Comment (borecjudo2) [reply src/main/java/com/sedliarov/learningtable/service/GroupService.java]: And the same for controller layer and for other entities

File: src/main/java/com/sedliarov/learningtable/service/impl/GroupServiceImpl.java [hunk 36]
```diff
@@ -0,0 +1,53 @@
...
+
+    @Override
+    public void deleteGroup(UUID id) {
+        repository.deleteById(id);
+    }
+
+    @Override
+    public GroupDto getGroupById(UUID id) {
+        return mapper.entityToDto(repository.findById(id).orElseThrow(EntityNotFoundException::new));
+    }
```
Comment (borecjudo2) [inline src/main/java/com/sedliarov/learningtable/service/impl/GroupServiceImpl.java]: Maybe add some message for exception?

File: src/main/java/com/sedliarov/learningtable/service/impl/StudentServiceImpl.java [hunk 37]
```diff
@@ -0,0 +1,53 @@
...
+
+    @Override
+    public void deleteStudent(UUID id) {
+        repository.deleteById(id);
+    }
+
+    @Override
+    public StudentDto getStudentById(UUID id) {
+        return mapper.entityToDto(repository.findById(id).orElseThrow(EntityNotFoundException::new));
+    }
```
Comment (borecjudo2) [inline src/main/java/com/sedliarov/learningtable/service/impl/StudentServiceImpl.java]: Maybe add some message for exception?

File: src/main/java/com/sedliarov/learningtable/service/impl/TeacherServiceImpl.java [hunk 38]
```diff
@@ -0,0 +1,53 @@
...
+
+    @Override
+    public void deleteTeacher(UUID id) {
+        repository.deleteById(id);
+    }
+
+    @Override
+    public TeacherDto getTeacherById(UUID id) {
+        return mapper.entityToDto(repository.findById(id).orElseThrow(EntityNotFoundException::new));
+    }
```
Comment (borecjudo2) [inline src/main/java/com/sedliarov/learningtable/service/impl/TeacherServiceImpl.java]: Maybe add some message for exception?

File: src/main/java/com/sedliarov/learningtable/controller/GroupController.java [hunk 39]
```diff
@@ -0,0 +1,49 @@
...
+    @ResponseStatus(HttpStatus.NO_CONTENT)
+    public GroupDto updateGroup(@PathVariable UUID id, @RequestBody GroupDto groupDto) {
+        return service.updateGroup(id, groupDto);
+    }
+
+    @GetMapping("/{id}")
+    @ResponseStatus(HttpStatus.OK)
+    public GroupDto getGroup(@PathVariable UUID id) {
+        return service.getGroupById(id);
+    }
```
Comment (borecjudo2) [inline src/main/java/com/sedliarov/learningtable/controller/GroupController.java]:
```suggestion
    public GroupDto getGroupById(@PathVariable UUID id) {
        return service.getGroupById(id);
    }
```

File: src/main/java/com/sedliarov/learningtable/controller/StudentController.java [hunk 40]
```diff
@@ -0,0 +1,47 @@
...
+    @ResponseStatus(HttpStatus.NO_CONTENT)
+    public StudentDto updateStudent(@PathVariable UUID id, @RequestBody StudentDto studentDto) {
+        return service.updateStudent(id, studentDto);
+    }
+
+    @GetMapping("/{id}")
+    @ResponseStatus(HttpStatus.OK)
+    public StudentDto getStudent(@PathVariable UUID id) {
+        return service.getStudentById(id);
+    }
```
Comment (borecjudo2) [inline src/main/java/com/sedliarov/learningtable/controller/StudentController.java]:
```suggestion
    public StudentDto getStudentById(@PathVariable UUID id) {
        return service.getStudentById(id);
    }
```

File: src/main/java/com/sedliarov/learningtable/controller/TeacherController.java [hunk 41]
```diff
@@ -0,0 +1,48 @@
...
+    @ResponseStatus(HttpStatus.NO_CONTENT)
+    public TeacherDto updateTeacher(@PathVariable UUID id, @RequestBody TeacherDto teacherDto) {
+        return service.updateTeacher(id, teacherDto);
+    }
+
+    @GetMapping("/{id}")
+    @ResponseStatus(HttpStatus.OK)
+    public TeacherDto getTeacher(@PathVariable UUID id) {
+        return service.getTeacherById(id);
+    }
```
Comment (borecjudo2) [inline src/main/java/com/sedliarov/learningtable/controller/TeacherController.java]:
```suggestion
    public TeacherDto getTeacherById(@PathVariable UUID id) {
        return service.getTeacherById(id);
    }
```

**Review by borecjudo2 — APPROVED** [review]

**Pull request merged**